src/wallet/coinselection.cpp

```diff
@@ -488,14 +491,22 @@ CAmount GenerateChangeTarget(const CAmount payment_value, const CAmount change_f
     }
 }
 
+void SelectionResult::SetBumpFeeDiscount(const CAmount discount)
+{
+    // Overlapping ancestry can only lower the fees, not increase them
+    assert (discount >= 0);
+    bump_fee_group_discount = discount;
+}
+
+
 void SelectionResult::ComputeAndSetWaste(const CAmount min_viable_change, const CAmount change_cost, const CAmount change_fee)
 {
     const CAmount change = GetChange(min_viable_change, change_fee);
 
     if (change > 0) {
```

Comment:
nit: this `if` can be removed and instead we can check if change exist directly in `GetChange()`

Reply:
I started with that, and then realized that `ComputeAndSetWaste` and `GetSelectionWaste` now live in the same class, so the former essentially wrapping the latter doesn’t make sense anymore. Then I cleaned that up, but that broke a bunch of the waste calculation tests. So, finally, I propose to clean all this up in a follow-up PR, which I will open shortly.

Reply:
sure, that's definitely cosmetics and can wait for a follow up 👍

Reply:
I’ve opened https://github.com/bitcoin/bitcoin/pull/28366 to collect these changes